src/inputs.rs

```diff
@@ -115,23 +126,24 @@ impl<B: Backend> CompiledCircuit<B> {
                 let struct_info = self
                     .circuit
                     .struct_info(&qualified)
-                    .expect("compiler bug: couldn't find struct given as input");
+                    .ok_or_else(|| ParsingError::MissingStructIdent(struct_name.to_string()))?;
```

Comment:
that's an example of an error that would benefit from just being an `UnexpectedError` here. In general the flow should be this:

* any panic in code should return an `UnexpectedError` to the user
* if the user reports such an error, we know that we either have a bug (we assumed that this code should never be reached, but it was reached) or we have an opportunity to write a better error for that specific error

Reply:
I agree. I did it this way because the function returns ParsingError. I added some extra context in the message so we know it's a bug when the user encounters it